test/plugin_functional/plugins/core_provider_plugin/types.ts

```diff
@@ -16,22 +16,22 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
-import { CoreSetup, Plugin } from 'kibana/public';
+import { LegacyCoreSetup, LegacyCoreStart } from 'kibana/public';
 
 declare global {
   interface Window {
-    uiSettingsPlugin?: Record<string, any>;
-    uiSettingsPluginValue?: string;
+    np: {
+      setup: {
+        core: LegacyCoreSetup;
+        plugins: Record<string, any>;
```

Comment:
does it affect types if don't import a file with a declaration? 
// src/core/public/core_system.ts
![2019-12-17_11-52-47](https://user-images.githubusercontent.com/3198181/70989402-dad00980-20c3-11ea-9574-3d7c62605273.png)

I'm not oposed to cast type manually tho, since it's more explicit.